### ISSUE TITLE
Rustc  gives out-of-date/misleading help on macro-backtrace

### ISSUE THREAD

**Issue by apopiak** [rust-lang/rust#70423]
rustc gives out-of-date/misleading help on macro-backtrace.

[This PR](https://github.com/rust-lang/rust/commit/f6fc80206e9600ae753cdfbd762fb982afca48b0) seems to have changed the flag (from `external-macro-backtrace` to `macro-backtrace`), but not updated the help.

I tried this code:
```
decl_module! {
	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
		/// The minimum percentage to pay for a bond.
		const MinimumBondPrice: Perbill = T::MinimumBondPrice::get();
		// elided ...
	}
}
```

I expected to see this happen:
Warning that I have missing docs with a note on how to achieve this when it occurs in a macro.

Instead, this happened:
Warning with a misleading help note:
```
warning: missing documentation for a struct
   --> src/lib.rs:348:1
    |
348 | / decl_module! {
349 | |     pub struct Module<T: Trait> for enum Call where origin: T::Origin {
350 | |         /// The minimum percentage to pay for a bond.
351 | |         const MinimumBondPrice: Perbill = T::MinimumBondPrice::get();
...   |
428 | |     }
429 | | }
    | |_^
    |
    = note: this warning originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

## Meta

`rustc --version --verbose`:
```
rustc 1.44.0-nightly (f509b26a7 2020-03-18)
binary: rustc
commit-hash: f509b26a7730d721ef87423a72b3fdf8724b4afa
commit-date: 2020-03-18
host: x86_64-unknown-linux-gnu
release: 1.44.0-nightly
LLVM version: 9.0
```

**Comment by Centril** [comment]
cc @eddyb

**Comment by apopiak** [comment]
mmh, just checked out rust and it seems it's already fixed in `master`, just not in `nightly`

**Comment by apopiak** [comment]
aaand I think I might have only gotten the misleading warning on `stable`, not on `nightly` :grimacing:

**Comment by apopiak** [comment]
jep, confirmed that the confusing note was on stable, will close
